src/main/java/org/akaza/openclinica/service/ImportValidationServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+import core.org.akaza.openclinica.bean.odmbeans.ChildNoteBean;
+import core.org.akaza.openclinica.bean.odmbeans.DiscrepancyNoteBean;
+import core.org.akaza.openclinica.core.form.xform.QueryType;
+import core.org.akaza.openclinica.dao.hibernate.DiscrepancyNoteDao;
+import core.org.akaza.openclinica.dao.hibernate.UserAccountDao;
+import core.org.akaza.openclinica.domain.datamap.DiscrepancyNote;
+import core.org.akaza.openclinica.exception.OpenClinicaSystemException;
+import core.org.akaza.openclinica.i18n.util.ResourceBundleProvider;
+import org.akaza.openclinica.web.restful.errors.ErrorConstants;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Locale;
+
+@Service("importValidationService")
+public class ImportValidationServiceImpl implements ImportValidationService{
+    @Autowired
+    private UserAccountDao userAccountDao;
+    @Autowired
+    private DiscrepancyNoteDao discrepancyNoteDao;
+
+    public static final String FAILED = "Failed";
+    public static final String QUERY_KEYWORD = "Query";
+    private boolean isQueryNewStatusValid = true;
+    private boolean isQueryUpdatedStatusValid = true;
+    private boolean isQueryClosedStatusValid = true;
+
+    public void validateQuery(DiscrepancyNoteBean discrepancyNoteBean){
+        isQueryNewStatusValid = true;
+        isQueryUpdatedStatusValid = true;
+        isQueryClosedStatusValid = true;
+
+        ResourceBundleProvider.updateLocale(Locale.ENGLISH);
+        if(!checkDiscrepancyNoteTypeValid(discrepancyNoteBean.getNoteType()))
+            throw new OpenClinicaSystemException(FAILED, ErrorConstants.ERR_DISCREPANCY_NOTE_TYPE_NOT_VALID);
+        if(discrepancyNoteBean.getDisplayId() != null) {
+            DiscrepancyNote discNote = discrepancyNoteDao.findByDisplayId(discrepancyNoteBean.getDisplayId());
+            if (discNote != null) {
+                setResolutionStatusForCheckingChildNotesValidity(discNote.getResolutionStatus().getName());
+            }
+        }
+
+        for(ChildNoteBean childNoteBean : discrepancyNoteBean.getChildNotes()){
+            if(!checkDiscrepancyNoteStatusValid(childNoteBean.getStatus()))
+                throw new OpenClinicaSystemException(FAILED, ErrorConstants.ERR_DISCREPANCY_NOTE_STATUS_NOT_VALID);
+            if(!isUserExist(childNoteBean.getOwnerUserName()))
+                throw new OpenClinicaSystemException(FAILED, ErrorConstants.ERR_USER_NOT_VALID);
+            if(childNoteBean.getUserRef() != null && !isUserExist(childNoteBean.getUserRef().getUserName()))
+                throw new OpenClinicaSystemException(FAILED, ErrorConstants.ERR_ASSIGNED_USER_NOT_VALID);
+            if(StringUtils.isBlank(childNoteBean.getDetailedNote()))
+                throw new OpenClinicaSystemException(FAILED, ErrorConstants.ERR_DETAILED_NOTE_NOT_AVAILABLE);
```

Comment:
instead of throwing exception for each error, we should collect all all errors in a collection and throw the exception at the end with all errors.